core/test/usLDAPQueryTest.cpp

```diff
@@ -0,0 +1,123 @@
+
+#include "usTestUtils.h"
+#include "usTestingMacros.h"
+
+using namespace us;
+
+void TestLDAPQueryBundleMetaData(BundleContext* frameworkCtx)
+{
+    std::vector<std::shared_ptr<Bundle>> bundles = frameworkCtx->GetBundles();
+    LDAPFilter ldap( "(bundle.testproperty=YES)" );
+    
+    int actualMatchCount = 0;
+    int expectedMatchCount = 0;
+    
+    std::string bundleName;
+    Any bundleTestProperty;
+    
+    for (std::shared_ptr<Bundle> bundle : bundles)
+    {
+        bundleName = bundle->GetProperty("bundle.name").ToString();
+        bundleTestProperty = bundle->GetProperty("bundle.testproperty");
+        
+        if (!(bundleTestProperty.Empty()) && (bundleTestProperty.ToString().compare("YES") == 0))
+        {
+            expectedMatchCount++;
+        }
+        
+        if (ldap.Match(*bundle))
+        {
+            actualMatchCount++;
+        }
+    }
+    
+    std::string test_description = "Test LDAP query of bundle properties (expected # of matches: " + std::to_string(expectedMatchCount) + "  actual matches: " + std::to_string(actualMatchCount) + ")";
+    
+    US_TEST_CONDITION(actualMatchCount == expectedMatchCount, test_description);
+}
+
+void TestLDAPQueryServiceProperties(BundleContext* frameworkCtx)
+{
+    std::vector<std::shared_ptr<Bundle>> bundles = frameworkCtx->GetBundles();
+    LDAPFilter ldap( "(service.testproperty=YES)" );
+    
+    BundleContext* thisBundleCtx;
+    
+    int actualMatchCount = 0;
+    int expectedMatchCount = 1;
+
+    for (std::shared_ptr<Bundle> bundle : bundles)
+    {
+        
+        if (bundle->GetBundleId() != 0)
+        {
+            bundle->Start();
+            //std::cout << bundle->GetProperty("bundle.name").ToString() << std::endl;
+            thisBundleCtx = bundle->GetBundleContext();
+            ServiceReferenceU sr = thisBundleCtx->GetServiceReference("us::TestBundleLQService");
+        
+            if (sr && ldap.Match(sr))
+            {
+                actualMatchCount++;
+            }
+            bundle->Stop();
+        }
+    }
+    
+    std::string test_description = "Test LDAP query of service properties (expected # of matches: " + std::to_string(expectedMatchCount) + "  actual # of matches: " + std::to_string(actualMatchCount)+ ")";
+    
+    US_TEST_CONDITION(actualMatchCount == expectedMatchCount, test_description);
+}
+
+int usLDAPQueryTest(int /*argc*/, char* /*argv*/[])
+{
+    US_TEST_BEGIN("LDAPQueryTest");
+    
+    FrameworkFactory factory;
+    
+    std::shared_ptr<Framework> framework = factory.NewFramework(std::map<std::string, std::string>());
+    framework->Start();
+    
+    BundleContext* frameworkCtx = framework->GetBundleContext();
+    
+    InstallTestBundle(frameworkCtx, "TestBundleLQ_1");
+    InstallTestBundle(frameworkCtx, "TestBundleLQ_2");
+    InstallTestBundle(frameworkCtx, "TestBundleLQ_3");
+    InstallTestBundle(frameworkCtx, "TestBundleLQ_4");
+    InstallTestBundle(frameworkCtx, "TestBundleLQ_5");
```

Comment:
Why 5 bundles ? Consider adding the absolute minimum number of bundles required to satisfy the test cases